azure/services/managedclusters/managedclusters_test.go

```diff
@@ -21,234 +21,86 @@ import (
-				m.GetUserCredentials(gomockinternal.AContext(), "my-rg", "my-managedcluster").Return([]byte("credentials-user"), nil)
-				s.SetAdminKubeconfigData([]byte("credentials"))
-				s.SetUserKubeconfigData([]byte("credentials-user"))
-				s.SetKubeletIdentity("kubelet-id")
-				s.SetOIDCIssuerProfileStatus(nil)
-				s.SetOIDCIssuerProfileStatus(&infrav1.OIDCIssuerProfileStatus{
-					IssuerURL: ptr.To("oidc issuer url"),
-				})
-				s.UpdatePutStatus(infrav1.ManagedClusterRunningCondition, serviceName, nil)
+			Data: map[string][]byte{
+				secret.KubeconfigDataName: []byte("user credentials"),
 			},
-		},
-		{
-			name:          "fail to get managed cluster credentials",
-			expectedError: "error while reconciling adminKubeConfigData: failed to get credentials for managed cluster: internal server error",
-			expect: func(m *mock_managedclusters.MockCredentialGetterMockRecorder, s *mock_managedclusters.MockManagedClusterScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
-				s.ManagedClusterSpec().Return(fakeManagedClusterSpec)
-				r.CreateOrUpdateResource(gomockinternal.AContext(), fakeManagedClusterSpec, serviceName).Return(armcontainerservice.ManagedCluster{
-					Properties: &armcontainerservice.ManagedClusterProperties{
-						Fqdn:              ptr.To("my-managedcluster-fqdn"),
-						ProvisioningState: ptr.To("Succeeded"),
-					},
-				}, nil)
-				s.SetControlPlaneEndpoint(clusterv1.APIEndpoint{
-					Host: "my-managedcluster-fqdn",
-					Port: 443,
-				})
-				s.IsAADEnabled().Return(false)
-				s.AreLocalAccountsDisabled().Return(false)
-				m.GetCredentials(gomockinternal.AContext(), "my-rg", "my-managedcluster").Return([]byte(""), errors.New("internal server error"))
-			},
-		},
-	}
-
-	for _, tc := range testcases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			g := NewWithT(t)
-			t.Parallel()
-			mockCtrl := gomock.NewController(t)
-			defer mockCtrl.Finish()
-			scopeMock := mock_managedclusters.NewMockManagedClusterScope(mockCtrl)
-			credsGetterMock := mock_managedclusters.NewMockCredentialGetter(mockCtrl)
-			reconcilerMock := mock_async.NewMockReconciler(mockCtrl)
-
-			tc.expect(credsGetterMock.EXPECT(), scopeMock.EXPECT(), reconcilerMock.EXPECT())
-
-			s := &Service{
-				Scope:            scopeMock,
-				CredentialGetter: credsGetterMock,
-				Reconciler:       reconcilerMock,
-			}
-
-			err := s.Reconcile(context.TODO())
-			if tc.expectedError != "" {
-				g.Expect(err).To(HaveOccurred())
-				g.Expect(err).To(MatchError(tc.expectedError))
-			} else {
-				g.Expect(err).NotTo(HaveOccurred())
-			}
+		}
+		kclient := fakeclient.NewClientBuilder().
+			WithObjects(adminASOKubeconfig, userASOKubeconfig).
+			Build()
+		scope.EXPECT().GetClient().Return(kclient).AnyTimes()
+
+		scope.EXPECT().SetControlPlaneEndpoint(clusterv1.APIEndpoint{
+			Host: "fdqn",
+			Port: 443,
+		})
+		scope.EXPECT().ClusterName().Return(clusterName).AnyTimes()
```

Comment:
Do we know how many times ClusterName() is expected to be called? AnyTimes() will be true even if ClusterName() isn't called, so maybe it's better to use MinTimes()

Reply:
I don't think it's particularly important in this case that we record in the test exactly how many times this function should be called. Calls to this function I imagine will fluctuate over time more than most others as we add features and refactor, so keeping this a little vague seems okay to me.